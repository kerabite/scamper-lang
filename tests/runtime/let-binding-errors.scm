; let bindings telescope
(let
  ([x1 1]
   [y1 (+ x1 6)])
  (+ x1 y1))

; let bindings refer to future bindings

(let
  ([x2 y2]
   [y2 5])
  (+ x2 y2))

(let*
  ([x3 y3]
   [y3 5])
  (+ x3 y3))