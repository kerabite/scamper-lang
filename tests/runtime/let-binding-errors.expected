:10:3: Scope error:
    Variable x1 is not defined
    In program: x1
:7:9: Scope error:
    Variable y2 is not defined
    In program: y2
:7:14: Scope error:
    Variable y3 is not defined
    In program: y3
